app/controllers/landing_page_controller.rb

```diff
@@ -75,9 +75,13 @@ def resolve_link(type, id, normalized_data)
       if @link_resolvers[type].respond_to? :call
         @link_resolvers[type].call(type, id, normalized_data)
       else
-        normalized_data[type][id]
+        self.class.find_link(type, id, normalized_data)
       end
     end
+
+    def self.find_link(type, id, normalized_data)
+      normalized_data[type].find { |item| item["id"] == id }
+    end
```

Comment:
This is ok for now but let's see when the structure gets bigger if it needs a by-id index instead of linear search.